src/stream/src/executor/receiver.rs

```diff
@@ -131,3 +196,118 @@ impl Executor for ReceiverExecutor {
         &self.info.identity
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+    use std::sync::Arc;
+
+    use futures::{pin_mut, FutureExt};
+    use risingwave_common::array::StreamChunk;
+    use risingwave_pb::stream_plan::update_mutation::MergeUpdate;
+
+    use super::*;
+    use crate::executor::{ActorContext, Barrier, Executor, Mutation};
+    use crate::task::test_utils::{add_local_channels, helper_make_local_actor};
+
+    #[tokio::test]
+    async fn test_configuration_change() {
+        let schema = Schema { fields: vec![] };
+
+        let actor_id = 233;
+        let ctx = Arc::new(SharedContext::for_test());
+        let metrics = Arc::new(StreamingMetrics::unused());
+
+        // 1. Register info and channels in context.
+        {
+            let mut actor_infos = ctx.actor_infos.write();
+
+            for local_actor_id in [actor_id, 114, 514] {
```

Comment:
How about give variable names for 114 and 514 to improve readability, like `old_upstream_actor_id`, `new_upstream_actor_id`?